### PR TITLE
ci: optimize ParseLiveQuery.spec.js timeouts

### PR DESCRIPTION
**Pull request by FransGH**
### New Pull Request Checklist
- [X] I am not disclosing a [vulnerability](https://github.com/parse-community/parse-server/blob/master/SECURITY.md).
- [X] I am creating this PR in reference to an [issue](https://github.com/parse-community/parse-server/issues?q=is%3Aissue).

### Issue Description
Potential unnecessary long timeouts for the test.

Related issue: #7659

### Approach
Reduced timeouts resulting in 5x speedup.

### TODOs before merging
- [x] A changelog entry is created automatically using the pull request title (do not manually add a changelog entry)

### CODE REVIEW

**Comment by FransGH** [comment]
@mtrezza, after my last comment in the earlier PR I remained courious how much this could be further optimised. Hope this helps... (main change is the reduced tmo in `afterEach` from 1000 to 10msec)

File: spec/ParseLiveQuery.spec.js [hunk 1]
```diff
@@ -5,6 +5,8 @@ const validatorFail = () => {
   throw 'you are not authorized';
 };
 
+console.log('jasmine.ASYNC_TEST_WAIT_TIME=' + jasmine.ASYNC_TEST_WAIT_TIME);
```
Comment (FransGH) [inline spec/ParseLiveQuery.spec.js]: this needs to be removed...

File: spec/ParseLiveQuery.spec.js [hunk 2]
```diff
@@ -922,9 +918,9 @@ describe('ParseLiveQuery', function () {
     });
     await obj1.save();
     await Parse.User.logOut();
-    await new Promise(resolve => setTimeout(resolve, 200));
+    await new Promise(resolve => setTimeout(resolve, 100));
     await obj2.save();
-    await new Promise(resolve => setTimeout(resolve, 200));
+    await new Promise(resolve => setTimeout(resolve, 100));
```
Comment (FransGH) [inline spec/ParseLiveQuery.spec.js:921]: tried with 10msec but the test fails

File: spec/ParseLiveQuery.spec.js [hunk 3]
```diff
@@ -310,14 +312,10 @@ describe('ParseLiveQuery', function () {
...
       const original = req.original;
-
-      setTimeout(() => {
-        done();
-      }, 2000);
-
       if (current.get('foo') != original.get('foo')) {
         req.sendEvent = false;
       }
+      setTimeout(done, 1000);
```
Comment (FransGH) [inline spec/ParseLiveQuery.spec.js:316]: unclear why this is 1000msec, reducing will probably not cause the test to fail but might falsely have it succeed..

File: spec/ParseLiveQuery.spec.js [hunk 4]
```diff
@@ -733,9 +731,7 @@ describe('ParseLiveQuery', function () {
     });
     object.set({ foo: 'bar' });
     await object.save();
-    setTimeout(async () => {
-      done();
-    }, 1000);
+    setTimeout(done, 1000);
```
Comment (FransGH) [inline spec/ParseLiveQuery.spec.js:732]: unclear why this is 1000msec, reducing will probably not cause the test to fail but might falsely have it succeed..

File: spec/ParseUser.spec.js [hunk 5]
```diff
@@ -3960,18 +3960,21 @@ describe('Parse.User testing', () => {
...
     await user.destroy();
-    await new Promise(resolve => process.nextTick(resolve));
+    await new Promise(resolve => setTimeout(resolve, 10));
     for (const key of events) {
       expect(calls[key]).toHaveBeenCalled();
     }
+    subscription.unsubscribe();
+    const client = await Parse.CoreManager.getLiveQueryController().getDefaultLiveQueryClient();
+    client.close();
+    await new Promise(resolve => setTimeout(resolve, 10));
```
Comment (FransGH) [inline spec/ParseUser.spec.js]: this is the key change, was 1000msec

**Comment by FransGH** [comment]
Hi @dblythy, would you mind having a quick look at this? Anything in here that might negatively change the outcome of a test (especially false positives)?

**Comment by mtrezza** [comment]
I haven't looked at the individual changes, but generally I'd say:
- different timeout lengths may be required depending on the tests
- to approximate a min. timeout, I suggest running the test locally, which may be quicker than waiting for the GH CI to run

While this PR focuses on reducing the individual test run, there is another optimization perspective which is reducing the overall compute time. These two are currently the same, but they don't have to be. I opened https://github.com/parse-community/parse-server/issues/7672 for details.

**Comment by mtrezza** [comment]
Closing as the base branch of this PR `release-4.x.x` is a LTS branch that is no longer maintained. To further pursue this optimization please open a new PR with the `alpha` branch as the base branch.

**Pull request closed without merging**